airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -302,14 +302,15 @@ def __handle_failed_task(self, task_arn: str, reason: str):
                 self.__class__.MAX_RUN_TASK_ATTEMPTS,
                 task_arn,
             )
-            self.active_workers.increment_failure_count(task_key)
+            if increment:
+                self.active_workers.increment_failure_count(task_key)
```

Comment:
Now that I'm looking at it, shouldn't the task be removed from the `active_workers` queue if the task has failed? Especially if it is being retried, we add the same task to the `pending_tasks` queue, are we duplicating the task (because in the next `sync()` iteration, we will move the task from the `pending_tasks` queue to the `active_workers` queue)?

Reply:
Yeah, I agree with you, something here still doesn't look quite right. But maybe I'm missing something again.

Reply:
I had a chat with Syed and he has a better solution for this.  I'm going to cancel this PR and he will submit his solution shortly.